features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/resources/ResourcesDataStoreManager.kt

```diff
@@ -0,0 +1,164 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.resources
+
+import android.text.format.DateUtils
+import com.datadog.android.api.feature.FeatureScope
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.api.storage.datastore.DataStoreCallback
+import com.datadog.android.core.internal.persistence.Deserializer
+import com.datadog.android.core.persistence.Serializer
+import com.datadog.android.core.persistence.datastore.DataStoreContent
+import com.datadog.android.sessionreplay.internal.ResourcesFeature.Companion.SESSION_REPLAY_RESOURCES_FEATURE_NAME
+import java.util.Collections
+
+@Suppress("TooManyFunctions")
+internal class ResourcesDataStoreManager(
```

Comment:
> [!CAUTION] 
> Here we use two independent storages for the actual content and the lastModified information. 
> It could be the source of discrepancies and bugs, and IMO it would be better if the "lastModified" information was handled directly as part of the data store itself, or as part of the data being stored. Case in point, the "last modified" date is only written if : 
> - we fail to read the last modified date (`onFetchFailure` in the `initializeManager`)
> - there is no previous modified date (`onFetchLastUpdateDateSuccess`)
> - when the data is expired (`handleDataStoreExpired`)
> This would also prevent duplicating the read/write calls. and having to synchronize both data in all possible cases.

Reply:
I originally went with this because that was the ios implementation, but a single file is cleaner - I've modified accordingly